src/main/java/ch/njol/skript/expressions/ExprBeaconValues.java

```diff
@@ -0,0 +1,211 @@
+
+	enum BeaconValues {
+		PRIMARY("primary [beacon] effect"),
+		SECONDARY("secondary [beacon] effect"),
+		RANGE("[beacon] range"),
+		TIER("[beacon] tier");
+
+		private final String pattern;
+
+		BeaconValues(String pattern) {
+			this.pattern = pattern;
+		}
+	}
+
+	private static final boolean PAPER_EVENTS = Skript.classExists("com.destroystokyo.paper.event.block.BeaconEffectEvent");
+	private static final boolean PAPER_RANGE = Skript.methodExists(Beacon.class, "getEffectRange");
+	private static BeaconValues[] beaconValues = BeaconValues.values();
+
+	static {
+		String patternEnding = " of %blocks%";
+		if (PAPER_EVENTS)
+			patternEnding = " [of %blocks%]";
+		int size = beaconValues.length;
+		String[] patterns = new String[size * 2];
+		for (BeaconValues value : beaconValues) {
+			patterns[2 * value.ordinal()] = "%blocks%['s] " + value.pattern;
+			patterns[2 * value.ordinal() + 1] = "[the] " + value.pattern + patternEnding;
+		}
+
+		Skript.registerExpression(ExprBeaconValues.class, Object.class, ExpressionType.PROPERTY, patterns);
+	}
+
+	private BeaconValues valueType;
+	private boolean isEffect;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		valueType = beaconValues[(int) floor(matchedPattern/2)];
+		if (valueType == BeaconValues.RANGE && !PAPER_RANGE) {
+			Skript.error(valueType.pattern + " can only be used on Paper.");
+			return false;
+		}
+		if (exprs[0] != null) {
+			setExpr((Expression<Block>) exprs[0]);
+		} else {
+			if (!getParser().isCurrentEvent(PlayerChangeBeaconEffectEvent.class, BeaconEffectEvent.class, BeaconActivatedEvent.class, BeaconDeactivatedEvent.class)) {
+				Skript.error("There is no beacon in a " + getParser().getCurrentEventName() + " event.");
+				return false;
+			}
+			setExpr(new EventValueExpression<>(Block.class));
+		}
+		if (valueType == BeaconValues.PRIMARY || valueType == BeaconValues.SECONDARY)
+			isEffect = true;
+
+		return true;
+	}
+
+	@Override
+	protected Object @Nullable [] get(Event event, Block[] source) {
+		return get(source, new Getter<Object, Block>() {
+			@Override
+			public @Nullable Object get(Block block) {
+				Beacon beacon  = (Beacon) block.getState();
+				switch (valueType) {
+					case PRIMARY -> {
+						if (PAPER_EVENTS && event instanceof PlayerChangeBeaconEffectEvent changeEvent) {
+							return changeEvent.getPrimary();
+						} else if (beacon.getPrimaryEffect() != null) {
+							return beacon.getPrimaryEffect().getType();
+						}
+					}
+					case SECONDARY-> {
+						if (PAPER_EVENTS && event instanceof PlayerChangeBeaconEffectEvent changeEvent) {
+							return changeEvent.getSecondary();
+						} else if (beacon.getSecondaryEffect() != null) {
+							return beacon.getSecondaryEffect().getType();
+						}
+					}
+					case RANGE -> {return beacon.getEffectRange();}
+					case TIER -> {return beacon.getTier();}
```

Comment:
remove {} or format it with new lines

Reply:
Well I cant remove the `{}` since the prior two cases need them, and as the enhanced switch is preferred over regular and I see no real benefit of having these two take an extra line when being on the same line is fine.

Reply:
you can remove the return to get something like this, which is valid
![image](https://github.com/user-attachments/assets/c0fbc0e6-b612-4e85-b8b4-ccd1e10c01b2)